zetonteam/zeton_django
# feat: add poetry

This merge request adds support for [poetry](https://python-poetry.org/) package management tool. 

Using Poetry over `requirements.txt` offers several advantages:

- Advanced dependency resolver helps avoid conflicts and ensures compatibility.
- Generates a `poetry.lock` file for reproducible environments.
- Manages project metadata within a single `pyproject.toml` file.
- Simplifies common tasks with intuitive commands for dependency management.
- Allows specification of development dependencies separately from production ones.
- Supports PEP 517/518 standards for building Python projects.

I validated the backend app manually after building a new docker image.

Steps to test this locally:

1. checkout `add-poetry` branch
2. run `make build` command to build a new Docker image that contains dependencies managed by `poetry`
3. generate a token and verify endpoint operation using swagger-ui
